bundles/net.tourbook.device/src/net/tourbook/device/sporttracks/FitLogSAXHandler.java

```diff
@@ -448,13 +544,84 @@ private void finalizeTour() {
+            try {
+               final float val = Float.parseFloat(_currentActivity.customDataFields.get("Left Torque Effectiveness Avg. [%]")); //$NON-NLS-1$
+               tourData.setPower_AvgLeftTorqueEffectiveness(val);
+               _currentActivity.avgPowerLeftTorqueEff = val;
+            } catch (final Exception exc) {
+               StatusUtil.log("Left Torque Effectiveness Avg. [%]" + ": FitlogEx parse error", exc); //$NON-NLS-1$ //$NON-NLS-2$
+            }
+         }
+         if (_currentActivity.customDataFields.containsKey("Right Torque Effectiveness Avg. [%]")) { //$NON-NLS-1$
+            try {
+               final float val = Float.parseFloat(_currentActivity.customDataFields.get("Right Torque Effectiveness Avg. [%]")); //$NON-NLS-1$
+               tourData.setPower_AvgRightTorqueEffectiveness(val);
+               _currentActivity.avgPowerRightTorqueEff = val;
+            } catch (final Exception exc) {
+               StatusUtil.log("Right Torque Effectiveness Avg. [%]" + ": FitlogEx parse error", exc); //$NON-NLS-1$ //$NON-NLS-2$
+            }
+         }
+         if (_currentActivity.customDataFields.containsKey("Left Pedal Smoothness Avg. [%]")) { //$NON-NLS-1$
+            try {
+               final float val = Float.parseFloat(_currentActivity.customDataFields.get("Left Pedal Smoothness Avg. [%]")); //$NON-NLS-1$
+               tourData.setPower_AvgLeftPedalSmoothness(val);
+               _currentActivity.avgPowerLeftPedalSmooth = val;
+            } catch (final Exception exc) {
+               StatusUtil.log("Left Pedal Smoothness Avg. [%]" + ": FitlogEx parse error", exc); //$NON-NLS-1$ //$NON-NLS-2$
+            }
+         }
+         if (_currentActivity.customDataFields.containsKey("Right Pedal Smoothness Avg. [%]")) { //$NON-NLS-1$
+            try {
+               final float val = Float.parseFloat(_currentActivity.customDataFields.get("Right Pedal Smoothness Avg. [%]")); //$NON-NLS-1$
+               tourData.setPower_AvgRightPedalSmoothness(val);
+               _currentActivity.avgPowerRightPedalSmooth = val;
+            } catch (final Exception exc) {
+               StatusUtil.log("Right Pedal Smoothness Avg. [%]" + ": FitlogEx parse error", exc); //$NON-NLS-1$ //$NON-NLS-2$
+            }
+         }
+      }
+
+
+      if (_currentActivity.avgPowerBalance != Float.MIN_VALUE) {
+         tourData.setPower_PedalLeftRightBalance((int) _currentActivity.avgPowerBalance);
+      }
+      if (_currentActivity.powerNormalized != Float.MIN_VALUE && _currentActivity.powerIntensityFactor != Float.MIN_VALUE) {
+         tourData.setPower_FTP((int) (_currentActivity.powerNormalized / _currentActivity.powerIntensityFactor));
       }
 
       tourData.setImportFilePath(_importFilePath);
 
       tourData.setDeviceTimeInterval((short) -1);
 
-      if (_currentActivity.timeSlices.isEmpty()) {
+      if (_currentActivity.timeSlices.size() == 0) {
```

Comment:
Why did you revert to `.size()==0` ?
See my [PR](https://github.com/wolfgang-ch/mytourbook/pull/257) several weeks ago that explains why using isEmpty() is more performance than requesting the size per this [rule](https://rules.sonarsource.com/java/tag/clumsy/RSPEC-1155)

Reply:
@FJBDev 
I followed your recommandations and made the correction see: https://github.com/wolfgang-ch/mytourbook/pull/276